File: src/ExternalUI/Utility.cpp
/* 
 * File:   Utility.cpp
 * Author: christophe.delondre
 *
 * License: see LICENSE.xivap
 */

#include "Utility.h"

#include <sstream>

std::vector<std::string> Utility::str_explode(const std::string& s, char delim)
{
    std::vector<std::string> result;
    std::istringstream iss(s);

    for (std::string token; std::getline(iss, token, delim); )
    {
        result.push_back(std::move(token));
    }

    return result;
}

bool Utility::str_endswith(const std::string& fullString, const std::string& ending)
{
    if (fullString.length() >= ending.length()) 
    {
        return (0 == fullString.compare (fullString.length() - ending.length(), ending.length(), ending));
    } 
    else 
    {
        return false;
    }
}

File: src/ExternalUI/js/Utility.js
/*
 * File:   Utility.js
 * Author: christophe.delondre
 *
 * License: see LICENSE.xivap
 */

.pragma library

function validateIPAddress(address)
{
    return (/^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/.test(address));
}

function lpad(num, size)
{
    var s = num+"";
    while (s.length < size) s = "0" + s;
    return s;
}
